elm-canvas/elm-canvas.js

```diff
@@ -12,23 +12,38 @@ customElements.define(
       this.render();
     }
 
+    static get observedAttributes() {
+      return ['canvas.width', 'canvas.height'];
+    }
+
+
+    setCanvasDimensions(w, h){
+        var devicePixelRatio = window.devicePixelRatio || 1;
+        this.canvas.style.width = w;
+        this.canvas.style.height = h;
+        this.canvas.width = w * devicePixelRatio;
+        this.canvas.height = h * devicePixelRatio;
+        this.context.scale(devicePixelRatio, devicePixelRatio);
```

Comment:
```suggestion
        // Reset current transformation matrix to the identity matrix
        ctx.setTransform(1, 0, 0, 1, 0, 0);
        this.context.scale(devicePixelRatio, devicePixelRatio);
```

We should reset the transform matrix before re-setting scale to avoid unwanted side-effects on hi-dpi monitors.